static/app/components/searchSyntax/parser.tsx

```diff
@@ -666,6 +672,17 @@ export class TokenConverter {
     return null;
   };
 
+  /**
+   * Checks a filter against some non-grammar validation rules
+   */
+  checkFilterWarning = <T extends FilterType>(key: FilterMap[T]['key']) => {
+    if (key.type !== Token.KEY_SIMPLE) {
+      return null;
+    }
+    const keyName = getKeyName(key as TokenResult<Token.KEY_SIMPLE>);
+    return this.config.getFilterTokenWarning?.(keyName) ?? null;
```

Comment:
@evanpurkhiser I saw that `getKeyName` also supports `KEY_EXPLICIT_TAG` and `KEY_AGGREGATE`.
I am missing domain knowledge to fully understand those + afaik checking simple keys is enough for the current feature. Therefore, I would restrict this functionality to `Token.KEY_SIMPLE` for now. 🤔

Reply:
`KEY_EXPLICIT_TAG` looks like this: `tag[my_tag]:` which can also just be written as `my_tag` when it doesn't conflict with anything.

`KEY_AGGREGATE` is for things like `p99(transaction.duration):>50`

Reply:
Restricting to KEY_SIMPLE seems fime